tests/core/test_dataobject_filters.py

```diff
@@ -162,6 +171,22 @@ def test_clip_box(datasets):
     assert clp is not None
 
 
+@pytest.mark.parametrize('as_composite', [True, False])
+def test_clip_box_no_unused_points(as_composite):
+    mesh = pv.Cube()
+    mesh = pv.MultiBlock([mesh]) if as_composite else mesh
+    new_bounds = (
+        mesh.bounds.x_min,
+        mesh.bounds.x_max,
+        mesh.bounds.y_min,
+        mesh.bounds.y_max,
+        0.2,
+        0.4,
```

Comment:
I would like to clearly define what these magic number means.
```suggestion
        mesh.bounds.z_min + (mesh.bounds.z_max - mesh.bounds.z_min) * 7 / 10,
        mesh.bounds.z_min + (mesh.bounds.z_max - mesh.bounds.z_min) * 8 / 10,
```